### PR TITLE
fix(table permission): partition name shouldn't display in the table permission check window

### LINKED ISSUE #3343: [Bug]: Create a partition table without permission to change. When creating a partition, the partition name is considered as the table name
### ODC version

odc432

### OB version

*

### What happened?

image

### What did you expect to happen?

Just need the change permission of the original table

### How can we reproduce it (as minimally and precisely as possible)?

Create a partition table without permission to change and create a partition

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by zijiacj**
#### What type of PR is this?
type-bug

#### Which issue(s) this PR fixes:

Fixes #3343

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/session/util/DBSchemaExtractor.java [hunk 1]
```diff
@@ -203,6 +203,9 @@ public RelationFactor visitAlter_table_action(Alter_table_actionContext ctx) {
             if (ctx.RENAME() != null) {
                 return null;
             }
+            if (ctx.alter_partition_option() != null) {
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/session/util/DBSchemaExtractor.java:206]: how about creating table ddl?
```sql
create table tbl(
    id int(11)
) partition by range(id) (
    partition p1 values less than(100)
)
```

**Comment by yhilmare** [comment]
btw, this is not the sql check window, it is table permission check window actually

**Comment by yhilmare** [comment]
It is not partition table name, please use partition name instead

File: server/odc-service/src/main/java/com/oceanbase/odc/service/session/util/DBSchemaExtractor.java (hunk 1 above)
Comment (zijiacj) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/session/util/DBSchemaExtractor.java:216]: have done

**Comment by zijiacj** [comment]
> It is not partition table name, please use partition name instead

ok

**Comment by zijiacj** [comment]
> btw, this is not the sql check window, it is table permission check window actually

ok

File: server/odc-service/src/main/java/com/oceanbase/odc/service/session/util/DBSchemaExtractor.java [hunk 2]
```diff
@@ -198,11 +199,23 @@ private static class OBMySQLRelationFactorVisitor extends OBParserBaseVisitor<Re
 
         private final Set<DBSchemaIdentity> identities = new HashSet<>();
 
+        @Override
+        public RelationFactor visitCreate_table_stmt(Create_table_stmtContext ctx) {
+            if (ctx.partition_option() != null) {
+                addRelationFactor(MySQLFromReferenceFactory.getRelationFactor(ctx.relation_factor()));
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/session/util/DBSchemaExtractor.java:205]: why should we add relation factor here?

Comment (zijiacj) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/session/util/DBSchemaExtractor.java]: Skipping over the ‘partition’

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**